checker/src/org/checkerframework/checker/index/searchindex/SearchIndexAnnotatedTypeFactory.java

```diff
@@ -0,0 +1,117 @@
+ * methods. It is part of the Index Checker.
+ */
+public class SearchIndexAnnotatedTypeFactory extends BaseAnnotatedTypeFactory {
+
+    public final AnnotationMirror UNKNOWN, BOTTOM;
+
+    public SearchIndexAnnotatedTypeFactory(BaseTypeChecker checker) {
+        super(checker);
+        UNKNOWN = AnnotationUtils.fromClass(elements, SearchIndexUnknown.class);
+        BOTTOM = AnnotationUtils.fromClass(elements, SearchIndexBottom.class);
+        this.postInit();
+    }
+
+    /**
+     * Provides a way to query the Constant Value Checker, which computes the values of expressions
+     * known at compile time (constant prop + folding).
+     */
+    ValueAnnotatedTypeFactory getValueAnnotatedTypeFactory() {
+        return getTypeFactoryOfSubchecker(ValueChecker.class);
+    }
+
+    @Override
+    protected Set<Class<? extends Annotation>> createSupportedTypeQualifiers() {
+        return new LinkedHashSet<>(
+                Arrays.asList(
+                        SearchIndex.class,
+                        SearchIndexBottom.class,
+                        SearchIndexUnknown.class,
+                        NegativeIndexFor.class));
+    }
+
+    @Override
+    public QualifierHierarchy createQualifierHierarchy(
+            MultiGraphQualifierHierarchy.MultiGraphFactory factory) {
+        return new SearchIndexQualifierHierarchy(factory);
+    }
+
+    private final class SearchIndexQualifierHierarchy extends MultiGraphQualifierHierarchy {
+
+        public SearchIndexQualifierHierarchy(
+                MultiGraphQualifierHierarchy.MultiGraphFactory factory) {
+            super(factory);
+        }
+
+        @Override
+        public boolean isSubtype(AnnotationMirror subAnno, AnnotationMirror superAnno) {
+            if (AnnotationUtils.areSameByClass(superAnno, SearchIndexUnknown.class)) {
+                return true;
+            }
+            if (AnnotationUtils.areSameByClass(subAnno, SearchIndexBottom.class)) {
+                return true;
+            }
+            if (AnnotationUtils.areSameByClass(subAnno, SearchIndexUnknown.class)) {
+                return false;
+            }
+            if (AnnotationUtils.areSameByClass(superAnno, SearchIndexBottom.class)) {
+                return false;
+            }
+            // All annotations have arguments, because each is either NegativeIndexFor or
+            // SearchIndex. All NegativeIndexFor annotations are subtypes of SearchIndex
+            // annotations whose arguments contain their arguments.
+            if (AnnotationUtils.areSameByClass(subAnno, NegativeIndexFor.class)) {
+                List<String> superArrays =
+                        IndexUtil.getValueOfAnnotationWithStringArgument(superAnno);
+                List<String> subArrays = IndexUtil.getValueOfAnnotationWithStringArgument(subAnno);
+                return superArrays != null
+                        && subArrays != null
+                        && superArrays.containsAll(subArrays);
```

Comment:
This looks backwards to me.
A `@SI({a,b})` can be used wherever a `@SI(a)` is needed, but not the reverse.  Therefore, `@SI({a,b})` is a subtype, not a supertype, of `@SI(a)`.

Before proceeding any further, I propose that you write the manual section that explains the type system, its semantics, and shows the subtyping relationships.  Writing those and getting feedback on them is supposed to be the first step, before any other part of implementation.  Skipping that step has caused wasted time.